[vercel] Update setup docs for Datadog-owned Marketplace integration (#3026)
DataDog/integrations-extras: vercel/README.md

- View and parse your application logs using [Datadog's Log Management][5]
- See the number of requests and 4xx/5xx HTTP errors to your serverless applications and APIs running on Vercel
- Monitor frontend and [Vercel Functions][9] performance with [Datadog Synthetics][6]
- Trace requests across your Vercel Functions with [Datadog APM][14]

## Setup

The Datadog integration for Vercel is installed from the Vercel Marketplace.

### Install Datadog from the Vercel Marketplace

1. Sign in to Vercel, then open the [Datadog listing in the Vercel Marketplace][7] and select **Connect Account**.

> 2. Select the Vercel team to install the integration into.

Reviewer: ```suggestion
2. Select the Vercel team in which you want to install the integration.
```

> 3. Select which projects to monitor - either all projects or a specific subset.

Reviewer: ```suggestion
3. Select which projects to monitor: either all projects or a specific subset.
```

4. Select **Connect Account**. After Vercel provisions the integration, you are redirected to the **Configure Vercel for Serverless** page in Datadog.

### Configure the integration

> On the **Configure Vercel for Serverless** page, complete the following steps and select **Add Integration**:

Reviewer: Selecting "Add Integration" should be the final step below.
```suggestion
On the **Configure Vercel for Serverless** page, complete the following steps:
```

2. **Select API Key** - Select **Create New** to create a dedicated key for this integration, or choose an existing API key.

3. **Enable Vercel Log Drain for Log Management** - Toggle **Log Drain** on to forward logs from your Vercel apps to Datadog. Set the sampling percentage (1-100%) for the share of logs to forward.

4. **Set Log Sources** - Select which Vercel log sources to forward. At least one is required:
   - **Static**: CDN and static asset request logs
   - **Lambda**: Serverless Function execution logs
   - **Build**: Build output logs
   - **Edge**: Edge Function and Edge Middleware logs
   - **External**: Logs from external services routed through Vercel
   - **Firewall**: Vercel Firewall event logs

   **Note:** To ensure metrics are collected, Datadog recommends enabling **Lambda** logs and **Static** logs. Enabling **Build** logs is recommended if you want to collect build metrics.

5. **Set Log Environment** - Select **Production**, **Preview**, or both. At least one is required.

6. **Enable Vercel Trace Drain for APM** - Toggle **Trace Drain** on to forward traces from your Vercel apps to Datadog. Set the sampling percentage (1-100%).

> After you select **Add Integration**, the **Vercel Integration Setup** status page opens and confirms that data is flowing into Datadog. From there you can navigate to the Vercel Monitoring summary, the out-of-the-box dashboard, the Log Explorer, the Trace Explorer, and Real User Monitoring.

Reviewer: ```suggestion
7. Select **Add Integration**. The **Vercel Integration Setup** status page opens and confirms that data is flowing into Datadog. From there you can navigate to the Vercel Monitoring summary, the out-of-the-box dashboard, the Log Explorer, the Trace Explorer, and Real User Monitoring.
```

The Vercel integration collects logs from your Vercel Project using Vercel's [Log Drains][10] feature.

### Traces

> The Vercel integration collects traces from your Vercel Project using Vercel's [Trace Drains][13] feature.

Reviewer 2: why is this a separate section if we already call it out above?

Author: This is a list of all the data that we collect as part of the integration, so above is the setup instructions and this is saying that we collect traces
